src/parser/mod.rs

```diff
@@ -7156,6 +7156,25 @@ impl<'a> Parser<'a> {
                 partition,
                 with_name,
             }
+        } else if dialect_of!(self is ClickHouseDialect|GenericDialect)
```

Comment:
Could you take a look at adding a method to the [dialect trait](https://github.com/apache/datafusion-sqlparser-rs/blob/main/src/dialect/mod.rs#L63-L72) for this functionality? Also note that testing wise as a result the test in moving to [common.rs in this style](https://github.com/apache/datafusion-sqlparser-rs/blob/main/tests/sqlparser_common.rs#L2781)

Reply:
I changed it and moved the case to the new file.